brambling/views/orders.py

```diff
@@ -247,6 +247,22 @@ def dispatch(self, request, *args, **kwargs):
                                        organization__slug=kwargs['organization_slug'])
         if not self.event.viewable_by(request.user):
             raise Http404
+        if 'code' in kwargs:
+            # If an anonymous user is trying to access an anonymous
+            # order without a session, redirect them to the login
+            # page.
+            matching_order = Order.objects.filter(event=self.event,
+                                                  person__isnull=True,
+                                                  code=kwargs['code']).exists()
+            if (matching_order and not request.user.is_authenticated() and
+                not Order.objects._get_session_code(request, self.event)):
+                url = reverse('login')
```

Comment:
Actually, we should probably display a helpful message any time that someone accesses an order they shouldn't have access to, and only redirect to their own order summary if this is actually their order.